arroyo-worker/src/rate_limiter.rs

```diff
@@ -0,0 +1,26 @@
+use std::future::Future;
+use std::ops::Sub;
+use std::time::{Duration, Instant};
+
+pub struct RateLimiter {
```

Comment:
We don't generally move individual structs into their own modules unless they have very large implementations. Given this has just two methods, I think it fits better in lib.rs